pkg/prometheus/promcfg.go

```diff
@@ -2317,6 +2319,24 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 
 	if sc.Spec.RelabelConfigs != nil {
 		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
+	}
+
+	if shards != 1 {
+		alreadyHasShardLabel := false
+		if sc.Spec.RelabelConfigs != nil {
+			for _, relabelConfig := range sc.Spec.RelabelConfigs {
+				if relabelConfig.Action == "hashmod" {
```

Comment:
Hello, I have a suggestion to make the code more readable. 
Here you check the action of "hashmod" as the indicator of "has Shard Label or not"
Is it possible to change the target label "__tmp_hash"  to something like "__tmp_shard_hash"?
Then, you can check the target label "__tmp_shard_hash" and make sure that this is of the purpose for sharding. 
Also, it could also avoid misjudgment if somebody just add their own "hashmod" action for other purpose.

Reply:
@leviliangtw I'm not sure to understand your point about the "__tmp_shard_hash" label?

I'm not aware of other use cases for the "hashmod" action than target sharding so I'm fine with the current approach but we might want to convey the nuance in the docs (e.g. "if the relabelConfigs list contains an item with `action: hashmod` then the operator will not apply the automatic target sharding.").